spec/duck-model/duck_spec.rb

```diff
@@ -748,6 +752,33 @@
 
   end
 
+  describe "fetching rank for an instance" do
+    before {
+      [:quacky, :feathers, :wingy, :webby, :waddly, :beaky].each_with_index do |name, i|
+         Duck.where(id: @ducks[name].id).update_all(row: RankedModel::MAX_RANK_VALUE - i)
+         @ducks[name].reload
+      end
+    }
+
+    context {
+      subject { Duck.find_by(id: @ducks[:beaky]).row_rank }
+
+      it { should == 0 }
```

Comment:
I think at this point we need to decide if rank will be 0 or 1 based. Currently it looks like you have it as zero based, but I wonder if, since this is a a user-facing method, whether it'd make more sense for it to be 1 based?

Reply:
Upon thinking further, the existing ranking works on a 0 based index so it make sense for this to also return a 0 based rank :)